### PR TITLE
69 piece vision

### LINKED ISSUE #69: Piece Vision
- Function that shows what squares a piece can see (move to)
- Function that shows what a player can see with their pieces (think fog of war)

### PR DESCRIPTION
**Pull request by abrahimt**
Piece vision is obsolete but I added two useful functions.

### CODE REVIEW

File: src/board.rs [hunk 1]
```diff
@@ -115,6 +115,39 @@ impl Chessboard {
...
+            & cb.black_rooks
+            & cb.white_queen
+            & cb.black_bishops
+            & cb.black_king
+            & cb.black_knights
+            & cb.black_pawns
+            & cb.black_rooks
+            & cb.black_queen
+    }
+
```
Comment (chanson02) [inline src/board.rs]: simplify this to be `one_side_pieces(cb, true) | one_side_pieces(cb, false)`

**Review by chanson02 — CHANGES_REQUESTED** [review]
you thought

File: src/board.rs [hunk 2]
```diff
@@ -115,6 +115,39 @@ impl Chessboard {
...
+                & cb.white_queen
+        } else {
+            cb.black_bishops
+                & cb.black_king
+                & cb.black_knights
+                & cb.black_pawns
+                & cb.black_rooks
+                & cb.black_queen
+        }
+    }
```
Comment (chanson02) [inline src/board.rs]: is this really how rust likes to format things???? ew.

File: src/board.rs (hunk 1 above)
Comment (abrahimt) [reply src/board.rs]: Done

File: src/board.rs (hunk 2 above)
Comment (abrahimt) [reply src/board.rs]: Yes

File: src/board.rs [hunk 3]
```diff
@@ -115,6 +115,39 @@ impl Chessboard {
         self.white_turn = true;
     }
 
+    pub fn one_side_pieces(cb: &Chessboard, white: bool) -> u64 {
```
Comment (chanson02) [inline src/board.rs]: make  these take `&self` as parameters instead of chessboard objects.

**Review by chanson02 — CHANGES_REQUESTED** [review]
lmao good try

File: src/board.rs (hunk 3 above)
Comment (abrahimt) [reply src/board.rs]: done

**Review by chanson02 — APPROVED** [review]
I didn't even read it, I trust you

**Comment by chanson02** [comment]
@abrahimt it looks like you need to merge in main first.

**Pull request merged**